x-pack/plugins/fleet/server/services/output.ts

```diff
@@ -554,11 +555,25 @@ class OutputService {
     }
   ) {
     const originalOutput = await this.get(soClient, id);
-
-    if (originalOutput.is_preconfigured && !fromPreconfiguration) {
-      throw new OutputUnauthorizedError(
-        `Preconfigured output ${id} cannot be updated outside of kibana config file.`
-      );
+    if (originalOutput.is_preconfigured) {
+      if (!fromPreconfiguration) {
+        const allowEditFields = originalOutput.allow_edit ?? [];
+
+        const allKeys = Array.from(new Set([...Object.keys(data)])) as Array<keyof Output>;
+        const isNotNullOrUndefinedOrEmpty = (o: any) => o !== null && o !== undefined && o !== '';
```

Comment:
Just curious why you explicitly defined a function, aren't this all `falsy` values? It's not clear to me why you didn't check directly if the value is `truthy` instead

Reply:
Yes good call, I was only checking for null and undefined in my first iteration but I can probably replace this by a `truthy` expression, I am going to change that